garden_ai/globus_compute/remote_functions.py

```diff
@@ -15,8 +17,9 @@ def register_pipeline(
     container_uuid: str,
 ) -> str:
     try:
+        to_register = inject_env_kwarg(pipeline._composed_steps)
```

Comment:
nit that probably makes sense to return to in the MLFlow auth PR: I think the name `inject_env_kwarg` makes it seem like the injection is happening right now, at registration time. It seems like it actually makes it possible to inject stuff at execution time. I'd try to think of a name that conveys that.